dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -388,6 +393,62 @@ public boolean handleVanityURLRedirects(final VanityUrlRequestWrapper request, f
       return false;
   }
 
+    /**
+     * Encodes the redirect URL with the parameters from the request.
+     *
+     * @param uri          The URI to redirect to.
+     * @return The encoded redirect URL.
+     */
+    private String encodeRedirectURL(final String uri) {
+        try {
+            boolean hastProtocol = true;
+            String redirectURI = uri;
+            if (uri.startsWith("//")) {
+                hastProtocol = false;
+                redirectURI = "none:" + uri;
+            }
+            final URLUtils.ParsedURL urlToEncode = URLUtils.parseURL(redirectURI);
+            if (urlToEncode == null) {
+                Logger.warn(this, "Could not parse redirect URL: " + uri);
```

Comment:
A little concerned we may either be covering up an issue here that should be an exception and will go un noticed, or there is a valid case and we will be logging warnings that will be ignored or clutter the logs?  We should decide whether this is an error or not.

Reply:
Changed to throw exception for an invalid URL